assets/src/hooks/useApiCall.ts

```diff
@@ -0,0 +1,56 @@
+import { useEffect, useState } from "react"
+
+/**
+ * Return type of {@linkcode useApiCall}
+ */
+interface ApiLoadingResult<T> {
+  /**
+   * Indicates if {@linkcode useApiCall} is in the process of computing a new
+   * value
+   */
+  isLoading: boolean
+
+  /**
+   * The last value that {@linkcode useApiCall} resolved.
+   *
+   * Is `undefined` when it has not resolved or it resolves to `undefined`
+   */
+  result: T | undefined
```

Comment:
I could change this to conditionally return `result` only after it's loaded. 
```suggestion
  result?: T
```

That would then allow changing 
```
ln29: Promise<undefined | T>
```

to

```
ln29: Promise<T>
```

Reply:
I'd then need a sentinel value that isn't `undefined` for tracking the not loaded state, but I could take ideas from the result type in https://github.com/mbta/skate/pull/2531 to hide that value

Reply:
I'm conflicted about this. If we keep it exactly as is, I would recommend calling it `useCachedApiCall`, since it's doing an API call (actually it's doing ... anything that returns a promise?), and then caching the result through the next full API round-trip.

But I sort of think that that responsibility should live in the calling code instead. Let the API code be responsible for fetching and returning data, and if I want to keep track of a result after that, then it's on me, not on the API code.

And then the type I would want to see here is:

```
interface Loading {
  loading: true
}

interface Loaded<T> {
  loading: false
  result: T
}

type ApiCallResult<T> = Loading | Loaded<T>
```

(Yeah that does lead into the annoying moment where typescript won't let you load `acr.result` unless you've somehow convinced it that it's `Loaded` instead of `Loading`...)

Reply:
The annoying moment, which will be incredibly helpful in production code, but annoying in tests.

Reply:
We talked about keeping it as is, do you still want a rename to `useCachedApiCall` @joshlarson ?

Reply:
Nah I think this is fine. 

Can we explicitly have a separate state for `Loading` and `CachedLoading` though?

I think the actual thing that's throwing me off is your comment 👇 
```ts
/* 
 * Is `undefined` when it has not resolved or it resolves to `undefined`
 */
```

Having a separate type for `Loading` versus `CachedLoading` would allow for a distinction between "this is the first time we've used this API call" versus "we're loading, but the previously-cached value is `undefined`".

HOWEVER, we can proooooobably refactor to that later, so let's consider this comment non-blocking.

Reply:
I tried doing this and then all of a sudden tests started acting super weird 😅

Reply:
WELP! Guess that moves that into refactor-land then!

Reply:
Sorry - I meant follow-up-PR-refactor-land!